service/src/main/kotlin/xyz/lebkuchenfm/external/gemini/GeminiClient.kt

```diff
@@ -0,0 +1,139 @@
+package xyz.lebkuchenfm.external.gemini
+
+import com.github.michaelbull.result.Err
+import com.github.michaelbull.result.Ok
+import com.github.michaelbull.result.Result
+import dev.kord.rest.request.errorString
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.engine.okhttp.OkHttp
+import io.ktor.client.plugins.DefaultRequest
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.request.accept
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.http.ContentType
+import io.ktor.http.URLProtocol
+import io.ktor.http.appendPathSegments
+import io.ktor.http.contentType
+import io.ktor.http.isSuccess
+import io.ktor.serialization.kotlinx.json.json
+import io.ktor.server.config.ApplicationConfig
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import xyz.lebkuchenfm.domain.radiopersonality.llmprompts.LlmPromptRaw
+
+private val logger = KotlinLogging.logger {}
+
+class GeminiClient(config: ApplicationConfig) {
+    private val httpClient: HttpClient by lazy { prepareHttpClient() }
+
+    private val apiKey: String? by lazy {
+        config.propertyOrNull("${CONFIGURATION_KEY}.apiKey")?.getString()
+    }
+    private val model: String by lazy {
+        config.property("${CONFIGURATION_KEY}.model").getString()
+    }
+
+    suspend fun generateText(prompt: LlmPromptRaw): Result<String, GeminiGenerateTextError> {
+        apiKey ?: run {
+            logger.error { "Missing Google Gemini API key." }
+            return Err(GeminiGenerateTextError.ApiKeyMissing)
+        }
+
+        val requestBody = GenerateTextRequestBody(
+            systemInstruction = GenerateTextRequestBody.Content(
+                parts = listOf(GenerateTextRequestBody.Content.Part(prompt.systemPrompt)),
+            ),
+            contents = GenerateTextRequestBody.Content(
+                parts = listOf(GenerateTextRequestBody.Content.Part(prompt.prompt)),
+            ),
+            safetySettings = listOf(
+                GenerateTextRequestBody.SafetySetting("HARM_CATEGORY_HARASSMENT", "BLOCK_NONE"),
+                GenerateTextRequestBody.SafetySetting("HARM_CATEGORY_HATE_SPEECH", "BLOCK_NONE"),
+                GenerateTextRequestBody.SafetySetting("HARM_CATEGORY_SEXUALLY_EXPLICIT", "BLOCK_NONE"),
+                GenerateTextRequestBody.SafetySetting("HARM_CATEGORY_DANGEROUS_CONTENT", "BLOCK_NONE"),
+                GenerateTextRequestBody.SafetySetting("HARM_CATEGORY_CIVIC_INTEGRITY", "BLOCK_NONE"),
+            ),
+        )
+
+        val response = httpClient.post("v1beta/models") {
+            url {
+                appendPathSegments("$model:generateContent")
+                parameters.append("key", apiKey!!)
```

Comment:
apiKey could be added in `prepareHttpClient` and model declared in post constructor
```kotlin

        val response = httpClient.post("v1beta/models/$model:generateContent") {
            setBody(requestBody)
            contentType(ContentType.Application.Json)
            accept(ContentType.Application.Json)
        }
```

Reply:
Sure.